controllers/runnerpool_controller.go

```diff
@@ -263,3 +207,83 @@ func (r *RunnerPoolReconciler) reconcileDeployment(ctx context.Context, log logr
 
 	return nil
 }
+
+func (r *RunnerPoolReconciler) findRunnerContainer(d *appsv1.Deployment) *corev1.Container {
+	for i := range d.Spec.Template.Spec.Containers {
+		c := &d.Spec.Template.Spec.Containers[i]
+		if c.Name == constants.RunnerContainerName {
+			return c
+		}
+	}
+	return nil
+}
+
+func (r *RunnerPoolReconciler) addRunnerContainerIfNotExists(d *appsv1.Deployment) {
+	if c := r.findRunnerContainer(d); c != nil {
+		// When the runner container already exists, nothing to do.
+		return
+	}
+
+	// Create the runner container.
+	c := corev1.Container{
+		Name: constants.RunnerContainerName,
+	}
+	d.Spec.Template.Spec.Containers = append(d.Spec.Template.Spec.Containers, c)
+}
+
+func (r *RunnerPoolReconciler) makeRunnerContainerEnv(rp *actionsv1alpha1.RunnerPool) []corev1.EnvVar {
+	envs := []corev1.EnvVar{
+		{
+			Name: constants.PodNameEnvName,
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+		{
+			Name: constants.PodNamespaceEnvName,
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name:  constants.RunnerOrgEnvName,
+			Value: r.organizationName,
+		},
+		{
+			Name:  constants.RunnerRepoEnvName,
+			Value: rp.Spec.RepositoryName,
+		},
+		{
+			Name:  constants.RunnerPoolNameEnvName,
+			Value: rp.ObjectMeta.Name,
+		},
+	}
+
+	if rp.Spec.SlackAgentServiceName != "" {
+		envs = append(envs, corev1.EnvVar{
+			Name:  constants.SlackAgentEnvName,
+			Value: rp.Spec.SlackAgentServiceName,
+		})
+	}
+
+	// NOTE:
+	// We need not ignore the reserved environment variables here.
+	// Since the reserved environment variables are checked in the validating webhook.
+	envs = append(envs, rp.Spec.Template.Env...)
+
+	return envs
+}
+
+func (r *RunnerPoolReconciler) makeRunnerContainerPorts() []corev1.ContainerPort {
+	return []corev1.ContainerPort{
+		{
+			Protocol:      corev1.ProtocolTCP,
+			Name:          constants.RunnerMetricsPortName,
+			ContainerPort: constants.RunnerMetricsPort,
```

Comment:
This may cause infinite updates.
Because `targetPort` will be automatically filled.

Reply:
Thank you.
I checked the update operation, but the `targetPort` in Deployment was not filed.
(But some fields cause unnecessary update, so I fixed it.)

Reply:
I'm sorry.
I mistook it for Service Resource.